tket/src/Circuit/basic_circ_manip.cpp

```diff
@@ -33,27 +33,51 @@ namespace tket {
 // this method removes them and removes the vertices
 // from boundaries
 void Circuit::remove_blank_wires(bool keep_blank_classical_wires) {
-  VertexList bin;
-  unit_vector_t unused_units;
-  const Op_ptr noop = get_op_ptr(OpType::noop);
-  for (const BoundaryElement& el : boundary.get<TagID>()) {
-    if (!keep_blank_classical_wires || el.type() == UnitType::Qubit) {
-      Vertex in = el.in_;
-      Vertex out = el.out_;
-      VertexVec succs = get_successors(in);
-      if (succs.size() == 1 && succs.front() == out) {
-        dag[in].op = noop;
-        bin.push_back(in);
-        dag[out].op = noop;
-        bin.push_back(out);
-        unused_units.push_back(el.id_);
+  bool found_empty_bit_at_end = true;
+  while (found_empty_bit_at_end) {
```

Comment:
The logic here is quite hard to understand, and seems inefficient: we are looping over all the boundary elements multiple times. Is there a better way, perhaps iterating over registers and removing unused bits from each of them working backwards from the end?

Reply:
I have tried to do this in a different way to make it easier to understand. I think `find` is still iterating some part of the boundary, so I would assume that this is a bit less sufficient then the other option for large circuits.

Reply:
See [cc8d629](https://github.com/CQCL/tket/pull/1481/commits/cc8d6293e567307084a5887029219c795d29e9db)

Reply:
see @cqc-melf
[update loop over bitnames](https://github.com/CQCL/tket/pull/1481/commits/9193185e984d8d2a923bd1d60670f8c928d3d004)
[9193185](https://github.com/CQCL/tket/pull/1481/commits/9193185e984d8d2a923bd1d60670f8c928d3d004)